language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -523,33 +523,18 @@ impl LibraVM {
             txn_effects_to_writeset_and_events_cached(&mut (), effects)?;
 
         // Make sure epilogue WriteSet doesn't intersect with the writeset in TransactionPayload.
-        if !epilogue_writeset
-            .iter()
-            .map(|(ap, _)| ap)
-            .collect::<HashSet<_>>()
-            .is_disjoint(
-                &change_set
-                    .write_set()
-                    .iter()
-                    .map(|(ap, _)| ap)
-                    .collect::<HashSet<_>>(),
-            )
-        {
+        if !check_intersection(
+            epilogue_writeset.iter().map(|a| (&a.0, &a.1)),
+            change_set.write_set().iter().map(|a| (&a.0, &a.1)),
+        ) {
             let vm_status = VMStatus::Error(StatusCode::INVALID_WRITE_SET);
             return Ok(discard_error_vm_status(vm_status));
         }
-        if !epilogue_events
-            .iter()
-            .map(|event| event.key())
-            .collect::<HashSet<_>>()
-            .is_disjoint(
-                &change_set
-                    .events()
-                    .iter()
-                    .map(|event| event.key())
-                    .collect::<HashSet<_>>(),
-            )
-        {
+
+        if !check_intersection(
```

Comment:
I don't understand how this will prevent the duplicate reconfiguration events. For example:

```
epilogue.events: [Reconfig { seq: 1, time: 2}]
change_set.events: `[Reconfig { seq: 1, time: 2 }, Reconfig { seq: 2, time 3 }]
```

I don't think this can be prevented without hardcoding this check